polaris-core/src/main/java/org/apache/polaris/core/storage/StorageLocation.java

```diff
@@ -99,4 +111,23 @@ public boolean isChildOf(StorageLocation potentialParent) {
       return slashTerminatedLocation.startsWith(slashTerminatedParentLocation);
     }
   }
+
+  private static String scheme(String location) {
+    int schemePos = location.indexOf(':');
+    if (schemePos > 0) {
+      return location.substring(0, schemePos).toLowerCase(Locale.ROOT);
+    }
+    return null;
```

Comment:
Should we throw here?

Reply:
No - this is passed early in the making of the `StorageLocation` and can, as a result, get filesystem locations like `/var/sample/location`. This is how we're figuring out that this is a local file path.